src/services/build-service.ts

```diff
@@ -0,0 +1,50 @@
+import * as path from 'node:path'
+import * as fs from 'node:fs'
+import { CLIError } from '@oclif/errors'
+import debugFactory from './debug-factory-service'
+
+import { MICROSERVICES_FOLDER } from '../paths'
+import ProcessExecutorService from './process-executor-service'
+import { ComponentService } from './component-service'
+
+export class BuildService {
+  private static debug = debugFactory(BuildService)
+
+  public static async build(name: string): Promise<any> {
+    const componentService = new ComponentService()
+    const component = componentService.getComponent(name)
+
+    const { type, stack } = component
+
+    let componentPath: string
+    let buildCmd = ''
+    let buildCmdArgs: string[] = []
+
+    if (type === 'microservice' && stack === 'spring-boot') {
+      componentPath = path.resolve(MICROSERVICES_FOLDER, name)
+      buildCmd = 'mvn'
+      buildCmdArgs = ['clean', 'test']
+    } else {
+      throw new CLIError(`${stack} ${type} build not implemented`)
+    }
+
+    BuildService.debug(
+      `Building ${name} using ${buildCmd} ${buildCmdArgs.join(' ').trim()}`
+    )
+    BuildService.debug(`Component Path ${componentPath}`)
+
+    if (!fs.existsSync(componentPath)) {
+      throw new CLIError(`Directory ${componentPath} not exists`)
+    }
+
+    process.chdir(componentPath)
```

Comment:
I think we should move this to the ProcessExecutorService. wdyt @zonia3000 ?

Reply:
Options of the `executeProcess()` method accepts an optional parameter `workDir`, that is passed to the node `spawn` function. I think that should be the correct way to handle it, also considering when we will have to handle multiple processes in parallel, so each process may have a different working directory.

Reply:
Done